platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/IdentifierHighlighterPassFactory.java

```diff
@@ -23,7 +24,7 @@ public final class IdentifierHighlighterPassFactory {
   public IdentifierHighlighterPass createHighlightingPass(@NotNull PsiFile file,
                                                           @NotNull Editor editor,
                                                           @NotNull TextRange visibleRange) {
-    if (!editor.isOneLineMode() &&
+    if (!((EditorEx) editor).isEmbeddedIntoDialogWrapper() &&
```

Comment:
Now  multi-line editors contained in a dialog don't get identifier highlighting anymore. Perhaps the old and new check should be combined?